lib/shared/tlv/TlvBer.cxx

```diff
@@ -232,6 +232,44 @@ TlvBer::Builder::SetTag(uint8_t tag)
     return *this;
 }
 
+TlvBer::Builder&
+TlvBer::Builder::SetTag(uint16_t tag)
+{
+    std::array<uint8_t, sizeof tag> tagData;
+    std::memcpy(&tagData, &tag, sizeof tag);
+
+    if (std::endian::native == std::endian::big) {
+        TlvBer::Builder::SetTag(tagData);
+        return *this;
+    } else {
+        std::array<uint8_t, sizeof tag> tagDataRev;
+        std::transform(std::rbegin(tagData), std::rend(tagData), std::begin(tagDataRev), [](auto&& b) {
+            return b;
+        });
+        TlvBer::Builder::SetTag(tagDataRev);
+        return *this;
+    }
+}
+
+TlvBer::Builder&
+TlvBer::Builder::SetTag(size_t tag)
```

Comment:
Is this version actually needed anywhere? Alternatively, since unsigned types are implicitly convertible to `std::size_t`, could we only use this version and remove the `uint16_t` version?

Reply:
I don't think so. 
When I just had the uint16_t version the compiler complained about some file needing the size_t version. 
Regarding the other direction, size_t is always larger than uint16_t so if we enter a uint16_t as use it as a size_t, the resultant tag would have leading zero bytes, which is undesirable.